src/prettytable/prettytable.py

```diff
@@ -1511,6 +1511,28 @@ def clear(self) -> None:
     def copy(self):
         return copy.deepcopy(self)
 
+    def get_output(self, out_format: str = "text", **kwargs) -> str:
+        """Return string representation of specified format of table in current state.
+
+        Arguments:
+        out_format - resulting table format
+        kwargs - passed through to function that performs formatting
+        """
+        if out_format == "text":
+            return self.get_string(**kwargs)
+        if out_format == "html":
+            return self.get_html_string(**kwargs)
+        if out_format == "json":
+            return self.get_json_string(**kwargs)
+        if out_format == "csv":
+            return self.get_csv_string(**kwargs)
+        if out_format == "latex":
+            return self.get_latex_string(**kwargs)
+        raise ValueError(
+            f"Invalid format {out_format}. "
+            "Must be one of: text, html, json, csv, or latex"
+        )
```

Comment:
```suggestion
        supported_formats = {
            "text": self.get_string, 
            "html": self.get_html_string,
            "json": self.get_json_string,
            "csv": self.get_csv_string,
            "latex": self.get_latex_string,
        }
        
        if out_format in supported_formats:
            return supported_formats[out_format](self, **kwargs)

        raise ValueError(
            f"Invalid format {out_format}. "
            "Must be one of: text, html, json, csv, or latex"
        )
```

Let's try to minimize code repetition.

Reply:
I think the original is clearer in this case.

Reply:
@hugovk Yeah I agree, looks shorter but less readable now.
Thoughts on the latest version?

Reply:
I also think the original is clearer despite being a little longer. If this was a "table" that was going to grow much, I would be much more likely to endorse the more elegant solution.